docs/public-networks/how-to/upgrade-node.md

```diff
@@ -2,32 +2,245 @@
+for this upgrade script.
+You can also see CoinCashew for instructions on upgrading Besu by building from source.
+:::
+ 
+## Upgrade on Docker
+
+1. Update your Docker image:
+
+    ```bash
+    docker pull hyperledger/besu:latest
+    ```
+
+2. Stop the current container:
+
+    ```bash
+    docker stop besu-node
+    ```
+
+3. Start a new container with the updated image:
+
+    ```bash
+    docker run -d \
+      --name besu-node            \
+      -v besu-data:/opt/besu/data \
+      -v besu-config:/etc/besu    \
+      hyperledger/besu:latest
+    ```
 
-You can use the [Ansible role on Galaxy](https://galaxy.ansible.com/consensys/hyperledger_besu) directly or customize it to suit your needs.
+Here is an example `docker-compose.yml` file:
 
-Upgrade the Besu version on nodes by running the play with the new version. For more information, For more information, select **Read Me** on the [Ansible Galaxy Besu page](https://galaxy.ansible.com/consensys/hyperledger_besu).
+```yaml
+version: '3.8'
+services:
+  besu:
+    image: hyperledger/besu:latest
+    volumes:
+      - besu-data:/opt/besu/data
+      - besu-config:/etc/besu
+    ports:
+      - "8545:8545"
+      - "30303:30303"
+volumes:
+  besu-data:
+  besu-config:
+```
+
+## Upgrade on Kubernetes
+
+1. Update your deployment manifest with a new image version:
+
+    ```yaml
+    spec:
+      containers:
+      - name: besu
+        image: hyperledger/besu:new-version
+    ```
+
+2. Apply the update:
+
+    ```bash
+    kubectl apply -f besu-deployment.yaml
+    ```
+
+Here is an example PVC configuration:
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: besu-data
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Ti
+```
+
+## Upgrade on Ansible
+
+You can use the [Ansible role on Galaxy](https://galaxy.ansible.com/ui/standalone/roles/consensys/hyperledger_besu/)
+directly or customize it to suit your needs.
+
+Upgrade your Besu node by running the play with the new version.
+For more information, select **Documentation** on the [Ansible Galaxy Besu page](https://galaxy.ansible.com/ui/standalone/roles/consensys/hyperledger_besu/).
 
 The playbook:
 
 1. Stops Besu.
-1. Downloads the updated version.
-1. Applies any new configuration.
-1. Starts Besu.
+2. Downloads the updated version.
+3. Applies any new configuration.
+4. Starts Besu.
+
+## Verify post-upgrade
+
+### RPC methods
+
+If you have [JSON-RPC HTTP enabled](../reference/cli/options.md#rpc-http-enabled),
+you can use the following commands to verify that you've successfully upgraded your Besu node.
+
+Call [`eth_syncing`](../reference/api/index.md#eth_syncing) to check the node synchronization status:
+
+ ```bash
+ curl -X POST --data '{"jsonrpc":"2.0","method":"eth_syncing","params":[],"id":1}' http://127.0.0.1:8545
```

Comment:
Probably not necessary, but most of our curl requests now include the header `-H "Content-Type: application/json"`